### DISCUSSION TITLE
Creating a no redundancy pool for CloudNativePG - operator stops upgrading OSDs as they're not ready to stop

### DISCUSSION THREAD

**Discussion by oboudry-mvp** [rook/rook#14511] [Q&A] [answered]
I would like to deploy a postgresql database using the CloudNativePG operator. As the cluster of postgresql databases already manages replication, it's recommended to remove data redundancy at the storage level. 

For this I tried to create a pool with no redundancy using the CephBlockPool file below with "spec.replicated.size = 1"

```
apiVersion: ceph.rook.io/v1
kind: CephBlockPool
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"ceph.rook.io/v1","kind":"CephBlockPool","metadata":{"annotations":{},"name":"noreplicapool","namespace":"rook-ceph"},"spec":{"failureDomain":"host","replicated":{"requireSafeReplicaSize":false,"size":1}}}
  creationTimestamp: "2024-06-17T08:46:57Z"
  finalizers:
  - cephblockpool.ceph.rook.io
  generation: 2
  name: noreplicapool
  namespace: rook-ceph
  resourceVersion: "63373736"
  uid: 6b3692c0-7a89-4d16-9f8c-3605d5554b05
spec:
  application: ""
  erasureCoded:
    codingChunks: 0
    dataChunks: 0
  failureDomain: host
  mirroring: {}
  quotas: {}
  replicated:
    size: 1
  statusCheck:
    mirror: {}
status:
  info:
    failureDomain: host
    type: Replicated
  observedGeneration: 2
  phase: Ready

```

Following this I got a ceph status warning, and muted the POOL_NO_REDUNDANCY health check.

All worked fine until I tried to upgrade the rook-operator to the latest version, and noticed OSDs were not upgrading. I removed the noredundancy pool and it solved the issue.

That's certainly not the right way of doing things. Is there a "right" way of creating a pool without redundancy for situations where the redundancy will be managed by the pods and is not expected to be managed at the storage level ?

**Comment by travisn** [comment, marked as answer]
There are several upgrade settings to look at to handle this case when using replica 1 pools. With the default settings, the OSDs will not be upgraded as you are seeing. I'd recommend setting [continueUpgradeAfterChecksEvenIfNotHealthy: true](continueUpgradeAfterChecksEvenIfNotHealthy) and you might also want to reduce the [waitTimeoutForHealthyOSDInMinutes](https://github.com/rook/rook/blob/release-1.14/deploy/examples/cluster.yaml#L40-L45) since the default is 10 minutes. See also the [CephCluster settings](https://rook.io/docs/rook/latest/CRDs/Cluster/ceph-cluster-crd/#cluster-settings) for more details. Another option is `skipUpgradeChecks: true` for fastest upgrades, but that will affect all daemons, not just OSDs.

**Reply by oboudry-mvp** [reply]
Great, thanks for your help Travis ! I'll try with the first two. skipUpgradeChecks seems a bit too much.
